frontend/heir/mlir_emitter.py

```diff
@@ -36,6 +37,72 @@ def mlirType(numba_type: NumbaType) -> str:
   raise InternalCompilerError("Unsupported type: " + str(numba_type))
 
 
+def isIntegerLike(typ: NumbaType | MLIRType) -> bool:
+  if isinstance(typ, type) and issubclass(typ, MLIRType):
+    return typ in {I1, I8, I16, I32, I64}
+  if isinstance(typ, NumbaType):
+    return isinstance(typ, types.Integer) or isinstance(typ, types.Boolean)
+  raise InternalCompilerError(f"Encountered unexpected type {typ}")
+
+
+def isFloatLike(typ: NumbaType | MLIRType) -> bool:
+  if isinstance(typ, type) and issubclass(type, MLIRType):
+    return typ in {F32, F64}
+  if isinstance(typ, NumbaType):
+    return isinstance(typ, types.Float)
+  raise InternalCompilerError(f"Encountered unexpected type {type}")
+
+
+# Needed because, e.g. Boolean doesn't have a bitwidth
+def getBitwidth(typ: NumbaType | MLIRType) -> int:
```

Comment:
nit: the following may be more readable, and note the fix `type(type)` to `type(typ)`

```python
def getBitwidth(typ: NumbaType | MLIRType) -> int:
    match typ:
        case types.Integer(bitwidth=bw):
            return bw
        case types.Boolean():
            return 1
        case I1 | I8 | I16 | I32 | I64:
            # e.g., <class '...I32'>.__name__ -> "I32"
            return int(typ.__name__[1:])
        case _:
            raise InternalCompilerError(f"Unexpected type {typ} ({type(typ)})")
```

Same for `is*Like` (also note line 53 uses `type` instead of `typ`)